x/ccv/provider/keeper/relay.go

```diff
@@ -66,6 +66,17 @@ func (k Keeper) OnRecvVSCMaturedPacket(
 	// clean up index
 	k.DeleteUnbondingOpIndex(ctx, chainID, data.ValsetUpdateId)
 
+	// remove the VSC timeout timestamp for this chainID and vscID
+	var timestamp time.Time
+	k.IterateVscTimeoutTimestamps(ctx, chainID, func(ts time.Time, vscID uint64) bool {
```

Comment:
@mpoke This `(chainID, timestamp) -> vscID` indexing scheme is unintuitive. I think that it could be a lot easier to understand with a different way of indexing, but I'm not sure what that would be right now.

Reply:
It's the most efficient way I could think of to perform the following operations:
- add a VSC timeout timestamp for a VSCPacket with vscID
- timeout the VSCPackets, which requires iterating over timestamps
- remove a VSC timeout timestamp when receiving a VSCMaturedPacket with vscID
- remove all VSC timeout timestamp associated with a chainID when that consumer chain is removed

Reply:
Changed indexing from `(chainID, timestamp) -> vscID` to `(chainID, vscID) -> timestamp`.